docs/dev-guide/testing/unit-tests-common-practices.md

```diff
@@ -0,0 +1,120 @@
+# Unit Tests Common Practices
+
+## Naming Conventions
+
+### Test class
+
+Test class should follow the naming convention **`[ClassUnderTest]Tests`**.
+
+Example: The test class for a class named ProductController should be `ProductControllerTests`:
+
+```csharp
+[TestFixture]
+public class ProductControllerTests
+{
+    ...
+}
+```
+
+### Test method
+
+Test method should follow the naming convention **`[MethodUnderTest]_[BehaviourToTest]_[ExpectedResult]`**.
+
+Example: For a method called GetProduct, the behaviour that we want to test is that it should return an exsiting project.
+The name of the test should be `GetProduct_ProductExist_ProductReturned`:
+
+```csharp
+[Test]
+public async Task GetProduct_ProductExist_ProductReturned()
+{
+    ...
+}
+```
+
+## Unit Test Skeleton: Three Steps/Parts
+
+A unit test should be designed/cut into three steps:
+
+1. Arrange: The first part where input/expected data are defined
+2. Act: The second part where the behavior under test is executed
+3. Assert: The third and last part where assertions are done
+
+These three parts are visually defined with comments so that unit tests can be humain readeable:
+
+```csharp
+[Test]
+public async Task GetProduct_ProductExist_ProductReturned()
+{
+    // Arrange
+    var productId = Guid.NewGuid().ToString();
+    var expectedProduct = new Product
+    {
+        Id = productId
+    };
+
+    // Act
+    var product = this.productService.GetProduct(productId);
+
+    // Asset
+    _ = product.Should().BeEquivalentTo(expectedProduct);
+}
+```
+
+!!! Tip
+    On IoT Hub Portal, we use the library [fluentassertions](https://github.com/fluentassertions/fluentassertions) on unit tests for natural/humain readeable assertions.
+
+## Mock
+
+A unit test should only test its dedicated layer. Any lower layer that requires/interact with external resources should be mocked to make sure that unit tests are idempotents.
+
+!!! note
+    For example, we want to implement unit tests on a controller that requires three services. Each service depends on others services/repositories/http clients that require external resources like databases, APIs...
+    Any execution of unit tests that require on these external resources can be altered (not idempotent) because they depend on the uptime, the data of these resources.
```

Comment:
```suggestion
    Example: We want to implement unit tests for a controller that requires three services. Each service depends on other services/repositories/http clients that need external resources like databases, APIs.... 
    Any execution of unit tests that depend on these external resources can be modified (not idempotent) because they depend on the uptime and data of these resources.
```